source/frontend-packages.rst

```diff
@@ -0,0 +1,183 @@
+functionality in Python. You can also group related modules into a package, by placing
+them in a directory.
+
+You can import your module to load it from the ``index.js``
+
+.. code-block:: javascript
+
+    import './utils.js'
+
+The ``./`` is necessary to distinguish from modules that would be loaded from the
+standard library or NPM installed packages. These imports are simply relative paths to
+the location of the module.
+
+You only need to include modules you might consider "top-level" in ``index.js``. For
+example, if you defined a helper function ``get_csrf_token()`` in that ``utils.js``
+and only needed to use that one function in another module, you would import it there,
+
+.. code-block:: javascript
+
+    import get_csrf_token from 'utils.js';
+
+This would assume that, in ``utils.js``, the function had been exported.
+
+.. code-block:: javascript
+
+    export function get_csrf_token() {
+        ...
+    }
+
+Installing New Packages
+=======================
+
+When adding new frontend dependencies you should find an NPM packaged distribution
+of the version you need. Preferably the vendor or project will manage this, but the
+Javascript world is still catching up to proper packaging and you may find third-party
+distributions which are also acceptable.
+
+You can install an NPM package both locally and configured in your package.json
+with a single command::
+
+    npm install --save package-name==1.2.3
+
+Upgrading Existing Packages
+===========================
+
+Periodically we need to look at upgrading the versions of third-party packages we
+depend on. This includes multiple steps.
+
+* Identify new versions of packages we use.
+* Upgrade to a newer version of a package and successfully test that no regressions occur.
+* Update ``package.json`` with the new version and commit this change.
+
+We can make this process a little easier, and we can also enforce some rules about them
+thanks to NPMs. When specifying package versions in ``package.json`` we can tell NPM about
+how we want it to interpret the version number we give.
+
+* ``1.2`` installs the newest ``1.2`` patch release
+* ``1.2.x`` or ``~1.2.3`` do the same, but the later defines the original minimum specified
+  when the package version was last pinned.
+* ``1`` or ``1.x`` or ``^1.2.3`` specify a minimum version, but allow any more recent *minor*
+  release such as ``1.3.0`` to be installed, if it is available.
+
+NPM dictates packages follow strict semver versioning rules, which instruct that patch
+releases should only include bugfixes and minor releases can include new functionality but
+must remain backwards compatible. Only major releases are allowed to break backwards
+compatibility.
+
+These rules are followed pretty well, but there are instances when a project might fail to
+follow the "no breaking changes in minor releases" rule, either mistakenly or intentionally.
+A package should be pinned with the ``~`` constraint if it may do this.
```

Comment:
we still need to document it, I think, because its what installing with the `--save` flag will add to the project by default, and we should understand those implications at least, even if we try to use more explicit syntax when doing so manually.

The current `package.json` you'll get on a new project includes all packages pinned with the default `~` constraint.